ios/RCTMGL/RCTMGLMapView.m

```diff
@@ -78,6 +93,30 @@ - (void)setReactUserTrackingMode:(int)reactUserTrackingMode
     [self setUserTrackingMode:(NSUInteger)_reactUserTrackingMode animated:_animated];
 }
 
+#pragma mark - methods
+
+- (CLLocationDistance)getMetersPerPixelAtLatitude:(double)latitude withZoom:(double)zoomLevel
+{
+    double constrainedZoom = [[RCTMGLUtils clamp:[NSNumber numberWithDouble:zoomLevel]
+                                             min:[NSNumber numberWithDouble:self.minimumZoomLevel]
+                                             max:[NSNumber numberWithDouble:self.maximumZoomLevel]] doubleValue];
+    
+    double constrainedLatitude = [[RCTMGLUtils clamp:[NSNumber numberWithDouble:latitude]
+                                                 min:[NSNumber numberWithDouble:-LAT_MAX]
+                                                 max:[NSNumber numberWithDouble:LAT_MAX]] doubleValue];
+    
+    double constrainedScale = pow(2.0, constrainedZoom);
+    return cos(constrainedLatitude * DEG2RAD) * M2PI * EARTH_RADIUS_M / (constrainedScale * TILE_SIZE);
+}
+
+- (CLLocationDistance)altitudeFromZoom:(double)zoomLevel
+{
+    CLLocationDistance metersPerPixel = [self getMetersPerPixelAtLatitude:self.camera.centerCoordinate.latitude withZoom:zoomLevel];
+    CLLocationDistance metersTall = metersPerPixel * self.frame.size.height;
+    CLLocationDistance altitude = metersTall / 2 / tan(MGLRadiansFromDegrees(30) / 2.0);
+    return altitude * sin(M_PI_2 - MGLRadiansFromDegrees(self.camera.pitch)) / sin(M_PI_2);
```

Comment:
mapbox/mapbox-gl-native#5583 would be nice to have.

Reply:
Yes it would!